catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/TransactionRequestConverter.java

```diff
@@ -59,20 +57,78 @@
     }
 
     @Override
-    public void marshal(Object o, HierarchicalStreamWriter hierarchicalStreamWriter,
+    public void marshal(Object o, HierarchicalStreamWriter writer,
             MarshallingContext marshallingContext) {
+        if (o != null) {
+            if (CswTransactionRequest.class.isAssignableFrom(o.getClass())) {
+                CswTransactionRequest request = (CswTransactionRequest) o;
+
+                writer.addAttribute(CswConstants.SERVICE, request.getService());
+                writer.addAttribute(CswConstants.VERSION, request.getVersion());
+                writer.addAttribute(CswConstants.VERBOSE_RESPONSE,
+                        String.valueOf(request.isVerbose()));
+                writer.addAttribute(CswConstants.XMLNS + CswConstants.NAMESPACE_DELIMITER
+                        + CswConstants.CSW_NAMESPACE_PREFIX, CswConstants.CSW_OUTPUT_SCHEMA);
+                writer.addAttribute(CswConstants.XMLNS + CswConstants.NAMESPACE_DELIMITER
+                        + CswConstants.OGC_NAMESPACE_PREFIX, CswConstants.OGC_SCHEMA);
+                for (InsertAction insertAction : request.getInsertActions()) {
+                    writer.startNode("csw:Insert");
+                    writer.addAttribute(CswConstants.TYPE_NAME_PARAMETER,
+                            insertAction.getTypeName());
+                    marshallingContext.put(CswConstants.TRANSFORMER_LOOKUP_KEY,
+                            TransformerManager.ID);
+                    marshallingContext.put(CswConstants.TRANSFORMER_LOOKUP_VALUE,
+                            insertAction.getTypeName());
+                    for (Metacard metacard : insertAction.getRecords()) {
+                        marshallingContext.convertAnother(metacard, delegatingTransformer);
+                    }
+                    writer.endNode();
+                }
+                for (UpdateAction updateAction : request.getUpdateActions()) {
+                    writer.startNode("csw:Update");
+                    writer.addAttribute(CswConstants.TYPE_NAME_PARAMETER,
+                            updateAction.getTypeName());
+                    marshallingContext.put(CswConstants.TRANSFORMER_LOOKUP_KEY,
+                            TransformerManager.ID);
+                    marshallingContext.put(CswConstants.TRANSFORMER_LOOKUP_VALUE,
+                            updateAction.getTypeName());
+                    marshallingContext.convertAnother(updateAction.getMetacard(),
+                            delegatingTransformer);
+
+                    writer.endNode();
 
+                }
+                for (DeleteAction deleteAction : request.getDeleteActions()) {
+                    writer.startNode("csw:Delete");
+                    writer.addAttribute(CswConstants.TYPE_NAME_PARAMETER,
+                            deleteAction.getTypeName());
+                    writer.startNode("csw:Constraint");
+                    writer.addAttribute(CswConstants.VERSION, "2.0.0");
```

Comment:
I was matching this to the constraint version I found in the Itests. Should it be 2.0.2?